unit/solvers/smt2_incremental/smt_bit_vector_theory.cpp

```diff
@@ -166,3 +166,24 @@ TEST_CASE("SMT bit vector predicates", "[core][smt2_incremental]")
       smt_bit_vector_theoryt::signed_greater_than_or_equal(two, wider));
   }
 }
+
+TEST_CASE(
+  "SMT bit vector arithmetic operator implementation tests",
+  "[core][smt2_incremental]")
+{
+  const smt_bit_vector_constant_termt two{2, 8};
+  const smt_bit_vector_constant_termt three{3, 8};
+
+  SECTION("Addition")
+  {
+    const auto function_application =
+      smt_bit_vector_theoryt::addition(two, three);
+    REQUIRE(
+      function_application.function_identifier() ==
+      smt_identifier_termt("bvadd", smt_bit_vector_sortt{8}));
+    REQUIRE(function_application.get_sort() == smt_bit_vector_sortt{8});
+    REQUIRE(function_application.arguments().size() == 2);
+    REQUIRE(function_application.arguments()[0].get() == two);
+    REQUIRE(function_application.arguments()[1].get() == three);
+  }
```

Comment:
⛏️ We should test that calling `smt_bit_vector_theoryt::add` with bit_vectors of mismatched width or on bool sorted terms fails an invariant.